test/test_client.py

```diff
@@ -0,0 +1,77 @@
+from collections import Counter
+from unittest import TestCase
+import logging
+
+from mock import Mock, patch
+
+from apns2.client import APNsClient, CONCURRENT_STREAMS_SAFETY_MAXIMUM
+from apns2.payload import Payload
+
+class ClientTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        logging.basicConfig()
+        cls.tokens = ["%064x" % i for i in range(10000)]
+        cls.notification = Payload(alert="Hello, world!")
+        cls.topic = "com.example.App"
+
+    
+    def setUp(self):
+        self.open_streams = 0
+        self.max_open_streams = 0
+        self.mock_results = None
+        
+        with patch("apns2.client.HTTP20Connection") as mock_connection_constructor, \
+            patch("apns2.client.init_context"):
+            self.mock_connection = Mock()
+            self.mock_connection.get_response.side_effect = self.mock_get_response
+            self.mock_connection.request.side_effect = self.mock_request
+            self.mock_connection.remote_settings.max_concurrent_streams = 500
```

Comment:
This is a default that's overridden in `test_send_notification_batch_overrides_server_max_concurrent_streams_if_too_small` and `test_send_notification_batch_overrides_server_max_concurrent_streams_if_too_large`.